apps/linkedin-to-notion/src/components/LinkedInNotionSidePanel/Form.tsx

```diff
@@ -77,19 +76,53 @@ export const Form = ({
       }
     };
 
-    setLinkedinUrl(window.location.href.match(/https:\/\/[a-z]{2,4}\.linkedin\.com\/in\/[^/]+\//gim)[0]);
-
     setFormValues();
   }, [displayNotionValues, linkedinValues]);
 
+  const findProfileInNotionDatabase = async (databaseId: string, linkedinUrl: string): Promise<void> => {
```

Comment:
I'd add an idea of Notion loading here. With a state `isNotionLoading` set to true at the beginning of the function, false at the end and render the rest of the form (underneath the DB choice) only when the loading is done, What do you think?

Reply:
Agreed!